client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts

```diff
@@ -235,6 +237,16 @@ const siteMigration: Flow = {
 
 				case STEPS.PROCESSING.slug: {
 					if ( providedDependencies?.siteCreated ) {
+						if ( platformQueryParam && fromQueryParam ) {
+							return exitFlow(
+								'/setup/site-setup/' +
+									getFinalImporterUrl(
+										siteSlug,
+										fromQueryParam,
+										platformQueryParam as ImporterPlatform
+									)
+							);
+						}
```

Comment:
After looking at the implementation for `getFinalImporterUrl()` I have some concerns:
* That function can return URLs that would break if we prefix them with `/setup/site-setup`, depending on various feature flags etc
* We're not performing any sanity-checks on `platformQueryParam`, and I think we need to do that before we trigger a redirect
```suggestion
						if ( platformQueryParam && fromQueryParam ) {
							const isValidImporterEngine = getImporterEngines().includes( platformQueryParm );
							if ( isValidImporterEngine ) {
								const importerPath = getFinalImporterUrl(
									siteSlug,
									fromQueryParam,
									platformQueryParam as ImporterPlatform
								);
								// Make sure we're only allowing URLs for an importer* step
								if ( importerPath && importerPath.startsWith( 'importer' ) ) {
									return exitFlow( '/setup/site-setup/' + importerPath );
								}
							}
						}
```

Reply:
I added a couple of functions to help deal with this case.

https://github.com/Automattic/wp-calypso/pull/101513/commits/6bd5a0739d1dfcd943a964f3745a5d68b7b3422d